Gems/Atom/Feature/Common/Code/Source/SplashScreen/SplashScreenPass.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Memory/SystemAllocator.h>
+
+#include <AzCore/Component/TickBus.h>
+#include <Atom/RPI.Public/Pass/FullscreenTrianglePass.h>
+#include <Atom/RPI.Reflect/Pass/PassDescriptor.h>
+#include <Atom/Feature/SplashScreen/SplashScreenSettings.h>
+
+namespace AZ::Render
+{
+    class SplashScreenPass
+        : public AZ::RPI::FullscreenTrianglePass
+        , public AZ::TickBus::Handler
+    {
+        AZ_RPI_PASS(SplashScreenPass);
+
+    public:
+        AZ_RTTI(AZ::Render::SplashScreenPass, "{B12F4E30-94ED-4F69-A17D-85C65853ACD9}", AZ::RPI::FullscreenTrianglePass);
+        AZ_CLASS_ALLOCATOR(AZ::Render::SplashScreenPass, AZ::SystemAllocator, 0);
+        virtual ~SplashScreenPass();
+
+        static AZ::RPI::Ptr<SplashScreenPass> Create(const AZ::RPI::PassDescriptor& descriptor);
+
+        // TickBus overrides...
+        void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+
+        // Scope producer functions...
+        void SetupFrameGraphDependencies(AZ::RHI::FrameGraphInterface frameGraph) override;
+        void CompileResources(const AZ::RHI::FrameGraphCompileContext& context) override;
+
+    private:
+        SplashScreenPass(const AZ::RPI::PassDescriptor& descriptor);
+        void Clear();
+
+        // Pass behavior overrides...
+        void InitializeInternal() override;
+        void FrameBeginInternal(FramePrepareParams params) override;
+        void FrameEndInternal() override;
+
+        bool m_beginTimer = false;  // Flag to begin timing.
+        float m_screenTime = 10.0f; // How long the slash screen lasts. Will be initialized by setreg.
+        float m_lastRealTimeStamp;  // Time stamp in seconds, used to calculate unscaled delta time.
+
+        // Data struct passed to the shader.
+        struct SplashScreenParams
+        {
+            float m_fadingFactor;
+            uint32_t m_flags;
```

Comment:
these flags appear to be unused
```suggestion
```

Reply:
I'll remove it. I intended to add some control to render an image at the center with a fixed size.
Currently, the image will be stretched to fit the window size.

Reply:
oh, yeah a fixed image at center would be a nice option, although - being able to render a UICanvas would probably be the nicest option because then you could handle all the of the positioning, scaling even animating in the UICanvas - but I'm guessing this is before all of that is loaded?